### PR TITLE
Implement PartialEq for Value

### PR DESCRIPTION
**Pull request by imp**
Addresses #219

### CODE REVIEW

File: json/src/value.rs [hunk 1]
```diff
@@ -625,6 +625,137 @@ impl<I> ops::Index<I> for Value where I: Index {
...
+}
+
+
+impl PartialEq<Value> for String {
+    fn eq(&self, other: &Value) -> bool {
+        other.as_str().map_or(false, |s| s == self)
+    }
+}
+
+macro_rules! partialeq_ixx {
```
Comment (oli-obk) [inline json/src/value.rs]: you can merge all these macros, by passing the `as_*` function name and the `as *` cast type as macro arguments.

Comment (imp) [inline json/src/value.rs]: Right, good advice!

**Comment by imp** [comment]
@oli-obk Thanks for review. Indeed much cleaner now.

**Comment by oli-obk** [comment]
thanks

**Pull request merged**